### ISSUE TITLE
Color contrasts in the Backend should meet WCAG guidelines

### ISSUE THREAD

**Issue by blnkt** [ManifoldScholar/manifold#1126]
With a particular focus on text but also considering vital iconography and other interactive components whose functionality might be lost if not legible. Color contracts given in form FOREGROUND/BACKGROUND.

WCAG requires a contrast ratio of at least 4.5:1 for normal text and 3:1 for large text. Large text is defined as 14 point (typically 18.66px) and bold or larger, or 18 point (typically 24px) or larger. Recommendations are based on these minimum thresholds.

https://webaim.org/resources/contrastchecker/

BACKEND

1. Top Nav links #9a9a9a/#414042
![screen shot 2018-06-08 at 3 27 56 pm](https://user-images.githubusercontent.com/8166060/41184550-ee7e24a2-6b35-11e8-9673-600c7ea089f1.png)
Recommend #ADADAD/#414042

2. On Projects page "reset search" button #9a9a9a/#363636
![screen shot 2018-06-08 at 4 10 34 pm](https://user-images.githubusercontent.com/8166060/41184635-7ea2a5e4-6b36-11e8-8ff0-e744e09c4743.png)
Recommend #9E9E9E/#363636 or #d6d6d6/#363636

3. On Projects page projects listing heading and project subtitle #828282/#363636; edit button #696969/#363636
![screen shot 2018-06-08 at 4 18 29 pm](https://user-images.githubusercontent.com/8166060/41184832-d18262ee-6b37-11e8-83e0-dbf9e242197d.png)
Recommend #9E9E9E/#363636 or #d6d6d6/#363636

4. On Projects Page "Activity" listings #9a9a9a/#363636
![screen shot 2018-06-08 at 4 23 13 pm](https://user-images.githubusercontent.com/8166060/41184889-44f149d4-6b38-11e8-9ec1-14273294e1f0.png)
Recommend #9E9E9E/#363636 or #d6d6d6/#363636

5. Sidebar links #9a9a9a/#363636
![screen shot 2018-06-08 at 4 25 29 pm](https://user-images.githubusercontent.com/8166060/41184952-9e93f13a-6b38-11e8-9476-659137cd4534.png)
Recommend #9E9E9E/#363636

6. Form labels and placeholder text #696969/#363636 and instructions #9a9a9a/#363636
![screen shot 2018-06-08 at 4 26 50 pm](https://user-images.githubusercontent.com/8166060/41184972-c4d43fbc-6b38-11e8-8e84-5190cf460eaf.png)
Recommend #9E9E9E/#363636

7. Toggle instructions and labels #9a9a9a/#363636 and #696969/#363636
![screen shot 2018-06-08 at 4 27 19 pm](https://user-images.githubusercontent.com/8166060/41184984-dd57932c-6b38-11e8-9b5a-cc1a53c61b2a.png)
Recommend #9E9E9E/#363636

8. On Projects Page: General Thumbnail editor instructions and label #828282/#363636
![screen shot 2018-06-08 at 4 29 26 pm](https://user-images.githubusercontent.com/8166060/41185039-5799bdd6-6b39-11e8-8d54-535d838cb6fd.png)
Recommend #9E9E9E/#363636

9. List items' buttons #828282/#363636
![screen shot 2018-06-08 at 4 33 32 pm](https://user-images.githubusercontent.com/8166060/41185097-c1980af8-6b39-11e8-910e-838771074547.png)
Recommend #9E9E9E/#363636
![screen shot 2018-06-08 at 4 42 34 pm](https://user-images.githubusercontent.com/8166060/41185216-fef5df8c-6b3a-11e8-8bf7-a512e8caf470.png)
![screen shot 2018-06-08 at 4 43 03 pm](https://user-images.githubusercontent.com/8166060/41185221-07ea497a-6b3b-11e8-9076-48240899be60.png)

10. On Projects page Activity Events listings #9a9a9a/#363636
![screen shot 2018-06-08 at 4 35 04 pm](https://user-images.githubusercontent.com/8166060/41185113-00678290-6b3a-11e8-9f7e-4e72c88b3939.png)
Recommend #9E9E9E/#363636

11. On Projects: Activity events pagination #9a9a9a/#363636
![screen shot 2018-06-08 at 4 36 35 pm](https://user-images.githubusercontent.com/8166060/41185143-3b087684-6b3a-11e8-9fb6-b9d38c381fed.png)
Recommend #9E9E9E/#363636

12. On Projects: Log results text #9a9a9a/#363636
![screen shot 2018-06-08 at 4 38 14 pm](https://user-images.githubusercontent.com/8166060/41185166-76dffdc6-6b3a-11e8-9c3b-de389145117f.png)
Recommend #9E9E9E/#363636

13. Secondary nav breadcrumb and subtitle #9a9a9a/#363636; Buttons #696969/#363636
![screen shot 2018-06-08 at 4 41 07 pm](https://user-images.githubusercontent.com/8166060/41185205-e44ec43c-6b3a-11e8-9886-5ecb2c64b03f.png)
Recommend #9E9E9E/#363636

14. Code editor comments #BC9458/#363636; element #6C99BC/#363636; id #CC7833/#363636
![screen shot 2018-06-08 at 4 48 01 pm](https://user-images.githubusercontent.com/8166060/41185325-eff150b0-6b3b-11e8-8548-fb9047dc86f3.png)
Recommend comments #BF985F/#363636; element #7EA4C4/#363636; id #D59058/#363636

15. Injestion radios #9a9a9a/#363636
![screen shot 2018-06-08 at 4 52 13 pm](https://user-images.githubusercontent.com/8166060/41185353-5e029fdc-6b3c-11e8-9d11-ef6ac9c85636.png)
Recommend #9E9E9E/#363636

16. Resource kind-picker #9a9a9a/#363636
![screen shot 2018-06-08 at 4 53 47 pm](https://user-images.githubusercontent.com/8166060/41185373-98e3112c-6b3c-11e8-9189-b890511cf38a.png)
Recommend #9E9E9E/#363636

17. File uploader text #9a9a9a/#363636
![screen shot 2018-06-08 at 4 54 47 pm](https://user-images.githubusercontent.com/8166060/41185386-b84c79fe-6b3c-11e8-8ec0-ee94ba34e988.png)
Recommend #9E9E9E/#363636

18. Drawer text #696969/#222222
![screen shot 2018-06-08 at 4 56 52 pm](https://user-images.githubusercontent.com/8166060/41185410-f4911154-6b3c-11e8-8510-86d5d863cb10.png)
![screen shot 2018-06-08 at 4 57 10 pm](https://user-images.githubusercontent.com/8166060/41185412-0223a4ee-6b3d-11e8-82e1-8a109fbf9513.png)
![screen shot 2018-06-08 at 4 57 43 pm](https://user-images.githubusercontent.com/8166060/41185426-190c800e-6b3d-11e8-9a11-b707a39f331c.png)
Recommend #8A8A8A/#222222

19. Predictive text input in drawer #828282/#363636
![screen shot 2018-06-08 at 4 58 20 pm](https://user-images.githubusercontent.com/8166060/41185459-54edfb7a-6b3d-11e8-907c-034671a309f2.png)
Recommend #9E9E9E/#363636

**Comment by zdavis** [comment]
This is a place where we could definitely use some help if we're going to move forward, mostly in terms of coming up with a better format for Blake's research that we can act on.

Right now, all of our off-whites are declared in a single spot, which is rad, because it means changing this won't suck. That spot is here: https://github.com/ManifoldScholar/manifold/blob/master/client/src/theme/_variables.scss#L11

```
$neutral05: #f7f7f7;
$neutral10: #ececec;
$neutral20: #e6e6e6;
$neutral30: #d6d6d6;
$neutral40: #c3c3c3;
$neutral45: #b3b2b3;
$neutral50: #9a9a9a;
$neutral70: #828282;
$neutral75: #696969;
$neutral80: #555555;
$neutral85: #414042;
$neutral90: #363636;
$neutral95: #2e2e2e;
$neutral100: #262626;
$neutral110: #222222;
```
What would be _awesome_ is if somebody could go through Blake's recommendations and figure out what he's mapping each of these colors to, and give us a new version of this code that included higher contrast hex codes.

This will be complicated a little bit by the fact that Blake is in some places suggesting a color be changed to more than one color. For example, in one spot he recommends that #9A9A9A ($neutral50) be changed to #ADADAD. A little bit later, he suggests we change it to #9E9E9E. That kind of discrepancy will make this harder than it should be, and we could use some help sorting it out.

I would begin this process by recording all the changes Blake recommends. For example:

```
#9A9A9A / #414042    >    #ADADAD / #414042
#9A9A9A / #363636    >    #9E9E9E / #363636
```
Once we have that list, we can just make a decision to remove the discrepancies:
```
#9A9A9A / #414042    >    #ADADAD / #414042
#9A9A9A / #363636    >    #ADADAD / #363636
```
With this approach, we could reduce Blake's recommendations to a finite list of changes, and test each on for AA contrast (using, for example, https://webaim.org/resources/contrastchecker/)

From there, we could update the color variables in the sketch file to see how things look, and finally update the SASS variables as well.

If anyone on team Manifold wants to contribute to this effort, team CIC would be THRILLED to have the help.

**Comment by krosemichael** [comment]
Hi Zach - I'm happy to do some work on this. I'll start by recording the changes and then will post that here so we can resolve discrepancies together if that makes sense.

**Comment by zdavis** [comment]
That's great. Thank you! Let me know if you have any questions.

best,
Zach

**Comment by krosemichael** [comment]
Hi @zdavis - I'm not sure if this is what you're looking for. Let me know if not or next steps. 

#828282 / #363636	>	#9e9e9e / #363636 

#696969 / #363636	>	#d6d6d6 / #363636
#696969 / #363636	>	#9e9e9e / #363636 
#696969 / #222222	>	#8a8a8a / #222222 

#9a9a9a / #414042	>	#adadad / #414042 
#9a9a9a / #363636	>	#9e9e9e / #363636 
#9a9a9a / #363636       >      #d6d6d6 / #363636 
			
#bc9458 / #363636	>	#bf985f / #363636 
#6c99bc / #363636	>	#7ea4c4 / #363636 
#CC7833 / #363636	>	#d59058 / #363636

**Comment by Santoz2** [comment]
Hi, this is Zaijen, a Student Accessibility Assistant at UW.

I would also like to note that the color contrast on the Frontend should be reviewed as well.

This can be seen with the contrast between the "Enter Admin Mode" button and the background, which would be #52e3ac on #fff. Using https://webaim.org/resources/contrastchecker/ can help show that this creates a contrast ratio that does not meet WCAG standards.

**Comment by zdavis** [comment]
Hi @Santoz2 — see #1124 where we're discussing this.

**Comment by Santoz2** [comment]
Thank you @zdavis , I had made the mistake of only searching through currently open issues. Sorry that I did not catch that.

**Comment by zdavis** [comment]
No worries. Appreciate your help!
